src/main/java/org/broadinstitute/consent/http/resources/StudyResource.java

```diff
@@ -215,6 +215,11 @@ public Response updateStudyByRegistration(
     try {
       User user = userService.findUserByEmail(authUser.getEmail());
       Study existingStudy = datasetRegistrationService.findStudyById(studyId);
+      boolean canUpdateStudy = user.hasUserRole(UserRoles.ADMIN) ||
+          datasetService.isCreatorOrCustodian(user, existingStudy);
+      if (!canUpdateStudy) {
+        throw new NotFoundException("Study with ID " + studyId + " is not found");
```

Comment:
Wouldn't a `ForbiddenException` be more appropriate for this case?

Reply:
> Should swagger docs be updated for this case?

Good call, update added, thank you.

Reply:
Per standup, I'll move this to a 403